checked/ref.go

```diff
@@ -169,3 +171,37 @@ func (c *RefCount) DecWrites() {
 func (c *RefCount) NumWriters() int {
 	return int(atomic.LoadInt32(&c.writes))
 }
+
+// TrackObject sets up the initial internal state of the Ref for
+// leak detection.
+func (c *RefCount) TrackObject(v interface{}) {
+	if atomic.LoadUint64(&leakDetectionFlag) == 0 {
+		return
+	}
```

Comment:
May it be better to do this before the `TrackObject(...)` call rather than inside it?  Performing the extra call stack and also creating a stack based `interface{}` every time even when we don't actually want to perform any tracking seems like an expense we could avoid pretty easily.